Automattic/wp-calypso
# Plans: Remove PaymentMethods block

#### Changes proposed in this Pull Request

This removes the `PaymentMethods` component from calypso. Previously it was used to display a list of icons of available payment methods on the plans and Jetpack Connect pages, but it has to be hooked up to the shopping cart to do so, and as checkout and the shopping cart are undergoing a large refactoring it will be necessary to change this. To that end, I suggested removing this component entirely (as its design is quite old) and possibly replacing it in the future with something else (eg: a static list of accepted payment methods).

In particular, this allows us to remove `getEnabledPaymentMethods`: https://github.com/Automattic/wp-calypso/pull/47263

Plans Before:

<img width="1071" alt="Screen Shot 2020-11-06 at 7 13 00 PM" src="https://user-images.githubusercontent.com/2036909/98613945-64348f80-22c5-11eb-9036-3d12492aa87a.png">

Plans After:

<img width="1049" alt="Screen Shot 2020-11-09 at 8 10 10 PM" src="https://user-images.githubusercontent.com/2036909/98614971-9810b480-22c7-11eb-8621-f72e11cdaeb5.png">

Jetpack Connect Before:

<img width="771" alt="Screen Shot 2020-11-09 at 8 16 11 PM" src="https://user-images.githubusercontent.com/2036909/98615420-95fb2580-22c8-11eb-9014-73a9139a9bf0.png">

Jetpack Connect After:

<img width="729" alt="Screen Shot 2020-11-09 at 8 17 39 PM" src="https://user-images.githubusercontent.com/2036909/98615437-a3b0ab00-22c8-11eb-99ed-85f0fcd50c3f.png">

Depends on https://github.com/Automattic/wp-calypso/pull/47261
This is part of a chain of related PRs. The next one is https://github.com/Automattic/wp-calypso/pull/47263

#### Testing instructions

- Visit the plans page (`/plans`) and verify that (apart from this component being missing), nothing looks out-of-place.
- Visit Jetpack connect (`/jetpack/connect/plans`) and verify the same.
